lab/webapp/src/components/Dataset/components/DatasetMenu/components/Charts/BoxPlot/index.jsx

```diff
@@ -0,0 +1,199 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Header } from 'semantic-ui-react';
+import * as d3 from "d3";
+//import * as bananaSet from './bananaJson.json';
+
+class BoxPlot extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+
+    };
+  this.createBoxPlot = this.createBoxPlot.bind(this);
+  }
+
+  componentDidMount() {
+    this.createBoxPlot();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    //this.createBarGraph();
+  }
+
+  // from - https://www.d3-graph-gallery.com/graph/boxplot_basic.html
+  createBoxPlot(){
+    const { dataPreview, valByRowObj, rawKey, cleanKey } = this.props;
+    //window.console.log('datapreview for boxplot', dataPreview);
+    let margin = { top: 5, right: 60, bottom: 50, left: 85 },
+        width = 425 - margin.left - margin.right,
+        height = 300 - margin.top - margin.bottom;
+
+    // to make background of svg transparent set stroke & fill to none
+    // or do not specify background color
+    let svg = d3.select("#test_box_plot_" + cleanKey)
+    .append("svg")
+      .attr("width", width + margin.left + margin.right)
+      .attr("height", height + margin.top + margin.bottom)
+      .attr("stroke", "none")
+    .style("fill", "none")
+    .append("g")
+      .attr("transform",
+            "translate(" + margin.left + "," + margin.top + ")");
+    // get stats
+    let data_sorted = valByRowObj[rawKey].sort(d3.ascending);
```

Comment:
To my knowledge, d3.quantile depends on sorted data. Please make sure the data is properly sorted here.

Reply:
I believe that is the case as well. 

`valByRowObj` is a javascript object; it's the dataset with the column as a key and the value is an array with all of that columns values. The data is sorted using javascripts native array sort method with d3.ascending - https://github.com/d3/d3-array#ascending